swagger.json

```diff
@@ -6374,6 +6683,31 @@
             "in": "query",
             "type": "integer",
             "format": "int64"
+        },
+        "createUserParam": {
+            "name": "user",
+            "description": "User properties. Only one of organizationRole or tagRoles is required.",
+            "required": true,
+            "in": "body",
+            "schema": {
+                "type": "object",
+                "required": [
+                    "name",
+                    "email",
+                    "authType"
+                ],
+                "$ref": "#/definitions/UpsertUserBase"
```

Comment:
@silverspace When using `$ref` for the schema I am not seeing the required properties being marked in the UI. Is there a different way I should be handling required properties?

Reply:
I think the issue is that JSON schema treats the `allOf` as a literal concatenation of the values in the array, and the required attributes are not applied to the result. If you update the required attributes in the `#/definitions` section, they should show up as required. For example:

```
        "UserBase": {
            "type": "object",
            "required": [ "name", "email", "authType" ],
```

Reply:
This is caused by https://github.com/swagger-api/swagger-editor/issues/1664#issuecomment-372098616

A workaround is to use `allOf` to specify required attributes of a $ref:

```
        "createUserParam": {
            "name": "user",
            "description": "User properties. Only one of organizationRole or tagRoles is required.",
            "required": true,
            "in": "body",
            "schema": {
              "allOf": [
                  {
                     "$ref": "#/definitions/UpsertUserBase"
                  },
                ],
                "type": "object",
                "required": [
                    "name",
                    "email",
                    "authType"
                ],
            }
        },
```